sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/multi/MultiServiceEventClient.java

```diff
@@ -176,83 +164,74 @@ public OfferResponse offers(Collection<Protos.Offer> offers) {
+                        break;
+                    case REMOVE_CLIENT:
+                        // This service has completed uninstall and can be torn down.
+                        servicesToRemove.add(service);
+                        break;
+                    case START_UNINSTALL:
+                        // This service has completed running and can be switched to uninstall.
+                        servicesToUninstall.add(serviceName);
+                        break;
+                    }
                     break;
                 }
 
                 // If we run out of unusedOffers we still keep going with an empty list of offers.
                 // This is done in case any of the services depends on us to turn the crank periodically.
             }
-        } finally {
-            multiServiceManager.sharedUnlock();
-        }
 
-        // Decline short if any service isn't ready.
-        boolean anyServicesNotReady = false;
-        List<OfferRecommendation> recommendations = new ArrayList<>();
-        if (!servicesToGiveOffers.isEmpty()) {
-            LOGGER.info("Sending offers to {} service{}:",
-                    servicesToGiveOffers.size(), servicesToGiveOffers.size() == 1 ? "" : "s");
-            List<Protos.Offer> remainingOffers = new ArrayList<>();
-            remainingOffers.addAll(offers);
-            for (AbstractScheduler service : servicesToGiveOffers) {
-                OfferResponse offerResponse = service.offers(remainingOffers);
-                recommendations.addAll(offerResponse.recommendations);
-                if (!remainingOffers.isEmpty() && !offerResponse.recommendations.isEmpty()) {
-                    // Some offers were consumed. Update what remains to offer to the next service.
-                    remainingOffers =
-                            OfferUtils.filterOutAcceptedOffers(remainingOffers, offerResponse.recommendations);
-                }
-                boolean readyForOffers = offerResponse.result == OfferResponse.Result.PROCESSED;
-                if (!offerResponse.recommendations.isEmpty() || !readyForOffers) {
-                    // Only log result when it's non-empty/unusual
-                    LOGGER.info("{} offer result: {}[{} recommendation{}], {} offer{} remaining",
-                            service.getServiceSpec().getName(),
-                            offerResponse.result,
-                            offerResponse.recommendations.size(),
-                            offerResponse.recommendations.size() == 1 ? "" : "s",
-                            remainingOffers.size(),
-                            remainingOffers.size() == 1 ? "" : "s");
-                }
-                if (!readyForOffers) {
-                    anyServicesNotReady = true;
-                }
+            if (allServicesIdle) {
+                // ALL services are idle, so we can tell upstream to suppress.
```

Comment:
I would just drop this comment. It's talking about an upstream effect outside the scope of this class.